### DISCUSSION TITLE
Helping with documentation

### DISCUSSION THREAD

**Discussion by johnhooks** [luisherranz/block-hydration-experiments#193] [Announcements]
Hello, I'm really excited to have found this project. @adamziel suggested I take a look after a discussion about block SSR and interactivity. I'm going to read through everything a familiarize myself in preparation to contribute. I am new to WordPress and the Block Editor and the issue of client-side interactivity has been my biggest question while trying to figure out block development. 

I've noticed there isn't much inline documentation. Would it be helpful if I did some work to fill out JSDocs while learning the project?

**Comment by adamziel** [comment]
Cc @luisherranz

**Comment by luisherranz** [comment]
> Would it be helpful if I did some work to fill out JSDocs while learning the project?

Oh, absolutely! That would be fantastic.

Feel free to open as many PRs as you want, and use this discussion to ask questions about each part of the code or even open new discussions if the topic is broad enough 🙂

Any of the contributors will happily explain how the different parts of the system work.

**Reply by AlphaZebra** [reply]
Hi, Robert here. Luis, we talked for a few minutes after your great workshop at WCEU. I'm hoping to build a couple things to get my feet wet with Interactivity over the next few weeks. I also have a lot of tech editorial experience, though, and would like to help on the docs as well. I'm thinking it might make sense to write a beginner-friendly guide to what's going on, how it works, etc. Something targeted at WordPress developers who don't necessarily have experience with React and react-like development frameworks. @johnhooks: Maybe we can coordinate on some stuff...

**Reply by luisherranz** [reply]
That would be fantastic. Thanks, Robert! ❤️ 

Would you mind opening a discussion in the [new Interactivity API category of the Gutenberg repository](https://github.com/WordPress/gutenberg/discussions/categories/interactivity-api) to coordinate the efforts? I want to start promoting that place to hopefully centralize all the conversation. I still want to add a few more things (like a small roadmap) though.

**Comment by johnhooks** [comment]
@luisherranz awesome, I'm excited to get into it.

Question: I am playing with the movie example and I don't see any of the directives as attributes in the rendered markup, are they removed on the server?

Edit: I found some in the markup on the individual movie pages. The original question was about the ♥️ icons on the homepage... Hum I see them all now... I don't know what was happening, but initially I didn't see the `wp-*` directive attributes in the markup.

**Reply by luisherranz** [reply]
Right now we're somehow absorbing them in the `Directives` component. See: https://github.com/WordPress/block-interactivity-experiments/blob/main-wp-directives-plugin/src/runtime/hooks.js#L43-L56

I'd like to review that logic at some point, though.

**Reply by johnhooks** [reply]
Okay, thanks for confirming I wasn't crazy! I would vote to leave them in the DOM, it helps visualize what is going on when in the element view of the dev tools.

**Reply by johnhooks** [reply]
Especially for anyone with AlpineJS or LiveWire experience.

**Reply by johnhooks** [reply]
Though this is coming from someone that doesn't understand the internals yet. Just viewing it from the outside.

**Reply by luisherranz** [reply]
Fair enough 🙂

**Comment by luisherranz** [comment]
I'm going to close this in favor of: 

- https://github.com/WordPress/gutenberg/discussions/51928

Please, use the new discussion for coordination.
